### PR TITLE
value and actor distribution networks

### PR DESCRIPTION
**Pull request by hnyu**
Issue #344 and #340

### CODE REVIEW

File: alf/networks/value_networks.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+import torch
+import torch.nn as nn
+
+from alf.networks import EncodingNetwork
+import alf.layers as layers
+from alf.tensor_specs import TensorSpec
+
+
+def _create_lstm_cell_state_spec(hidden_size, dtype):
```
[the remark this answers is not in the document]
Comment (hnyu) [reply alf/networks/value_networks.py]: I moved it to layers.py

File: alf/networks/value_networks.py [hunk 2]
```diff
@@ -23,25 +23,6 @@
 from alf.tensor_specs import TensorSpec
```
Comment (le-horizon) [inline alf/networks/value_networks.py]: Does this file also need to include
`from alf.layers import _create_lstm_cell_state_spec`
?

File: alf/networks/actor_distribution_networks.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        Args:
+            observation (torch.Tensor): consistent with `input_tensor_spec`
+            state: empty for API consistent with ActorRNNDistributionNetwork
+
+        Returns:
+            act_dist (torch.distributions): action distribution
+            state: empty
+        """
+        act_dist = self._projection_net(self._encoding_net(observation))
+        return act_dist, state
```
Comment (emailweixu) [inline alf/networks/actor_distribution_networks.py]: need to implement state_spec. Perhaps we should have a common base class Network which provides all the necessary interfaces.

File: alf/networks/actor_distribution_networks.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+        self._lstm_cell = torch.nn.LSTMCell(
+            input_size=self._encoding_net.output_size,
+            hidden_size=lstm_hidden_size)
+        self._state_spec = _create_lstm_cell_state_spec(
+            lstm_hidden_size, input_tensor_spec.dtype)
+
+        input_tensor_spec = TensorSpec((lstm_hidden_size, ),
+                                       input_tensor_spec.dtype)
+        self._after_lstm_encoding_net = EncodingNetwork(
```
Comment (emailweixu) [inline alf/networks/actor_distribution_networks.py]: Perhaps we can have a LSTMEncodingNetwork to do all these work.

File: alf/networks/value_networks.py (hunk 2 above)
Comment (hnyu) [reply alf/networks/value_networks.py]: layers._create_lstm_cell_state_spec is what I used in this file

Comment (le-horizon) [reply alf/networks/value_networks.py]: Ah, you are right.

**Comment by hnyu** [comment]
Resolve conflict in build.sh

**Pull request merged**